docs/core/project-sdk/msbuild-props-desktop.md

```diff
@@ -0,0 +1,195 @@
+---
+title: MSBuild properties for Microsoft.NET.Sdk.Desktop
+description: Reference for the MSBuild properties and items that are understood by the .NET Desktop SDK which includes WPF and WinForms.
+ms.date: 02/04/2021
+ms.topic: reference
+author: adegeo
+ms.author: adegeo
+ms.custom: updateeachrelease
+no-loc: ["App.xaml", "Application.xaml", "ApplicationDefinition", "Page", "EmbeddedResource", "Compile", "None"]
+---
+# MSBuild reference for .NET Desktop SDK projects
+
+This page is a reference for the MSBuild properties and items that you use to configure Windows Forms (WinForms) and Windows Presentation Foundation (WPF) projects with the .NET Desktop SDK.
+
+> [!NOTE]
+> This article documents a subset of the MSBuild properties for the .NET SDK as it relates to desktop apps. For a list of common .NET SDK-specific MSBuild properties, see [MSBuild reference for .NET SDK projects](msbuild-props.md). For a list of common MSBuild properties, see [Common MSBuild properties](/visualstudio/msbuild/common-msbuild-project-properties).
+
+## Enable .NET Desktop SDK
+
+To use WinForms or WPF, configure your project file.
+
+### .NET 5.0
+
+WinForms and WPF projects should specify the following settings in the project file:
+
+- Target the .NET SDK `Microsoft.NET.Sdk`. For more information, see [Project files](overview.md#project-files).
+- Set the `TargetFramework` to `net5.0-windows`.
+- Add a UI framework property:
+  - Set `UseWPF` to `true` to import and use WPF.
+  - Set `UseWindowsForms` to `true` to import and use WinForms.
+- (Optional) Set `OutputType` to `WinExe`. This produces an app as opposed to a library. To produce a library, omit this property.
+
+```xml
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <OutputType>WinExe</OutputType>
+    <TargetFramework>net5.0-windows</TargetFramework>
+
+    <UseWPF>true</UseWPF>
+    <!-- or -->
+    <UseWindowsForms>true</UseWindowsForms>
+  </PropertyGroup>
+
+</Project>
+```
+
+### .NET Core 3.1
+
+- Target the .NET SDK `Microsoft.NET.Sdk.WindowsDesktop`. For more information, see [Project files](overview.md#project-files).
+- Set the `TargetFramework` to `netcoreapp3.1`.
+- Set one of the following to enable a UI framework:
+  - Set `UseWPF` to `true` to use WPF.
+  - Set `UseWindowsForms` to `true` to use WinForms.
+- (Optional) Set `OutputType` to `WinExe`. This produces an app as opposed to a library. To produce a library, omit this property.
+
+```xml
+<Project Sdk="Microsoft.NET.Sdk.WindowsDesktop">
+
+  <PropertyGroup>
+    <OutputType>WinExe</OutputType>
+    <TargetFramework>netcoreapp3.1</TargetFramework>
+
+    <UseWPF>true</UseWPF>
+    <!-- or -->
+    <UseWindowsForms>true</UseWindowsForms>
+  </PropertyGroup>
+
+</Project>
+```
+
+## WPF default include exclude
```

Comment:
```suggestion
## WPF default includes and excludes
```

Reply:
So my reasoning on this: even though it's grammatically incorrect, it prevents the H2 TOC from wrapping and still communicates the thought. However, looking at the winforms and playing with the preview site, there is no way to use "Windows Forms" in the H2 and not prevent wrapping. Even using `WinForms includes and excludes` causes wrapping.. Not sure what a good H2 will end up being.

Any other ideas to prevent wrapping? Here are some tests I did: (including winforms stuff as hopefully I'll get this information eventually)

| Title | Single line H2 |
| --- | --- |
| WPF default include exclude | ✔️  |
| WPF default includes & excludes | ❌  |
| WPF default includes and excludes | ❌  |
| WPF includes and excludes | ✔️  |
| WPF includes & excludes | ✔️  |
| Windows Forms *  | ❌ |
| WinForms default include exclude | ❌ |
| WinForms default includes & excludes | ❌ |
| WinForms default includes and excludes | ❌ |
| WinForms includes and excludes | ❌ |
| WinForms includes & excludes | ✔️  |

*Any title with Windows Forms wrapped

The only thing I can see that creates a clean H2 TOC is 

- WPF includes & excludes
- WinForms includes & excludes

Reply:
Can we trick with non-breakable spaces (`&nbsp;`)?

Reply:
We could also consider adding WPF and Windows Forms as subsections on the [Overview page's includes/excludes section](https://docs.microsoft.com/en-us/dotnet/core/project-sdk/overview#default-includes-and-excludes). That would solve your H2 wrapping problem, and might make more sense because you wouldn't have to duplicate the includes/excludes table from that page (which might get out of date if the main article is updated).

Reply:
I think it makes more sense to just have all the information in this article. I don't think the defaults will change very often if ever.